src/components/interaction-designer/Connection.vue

```diff
@@ -140,18 +141,16 @@ export default {
 
     this.line = new LeaderLine(start, end, this.options)
 
-    // Add event listeners
-    const self = this
-    // the only way to identify current line so far: https://github.com/anseki/leader-line/issues/185
-    const connectionElement = document.querySelector('body>.leader-line:last-of-type')
+    this.connectionDomHandle = document.getElementById(`leader-line-${this.line._id}-line-path`).closest('svg.leader-line')
 
-    connectionElement.addEventListener('click', self.clickHandler, false)
+    // Add event listeners
+    this.connectionDomHandle.addEventListener('click', this.clickHandler, false)
 
-    connectionElement.addEventListener('click', self.clickAwayHandler(connectionElement), false)
+    this.connectionDomHandle.addEventListener('click', this.clickAwayHandler(this.connectionDomHandle), false)
```

Comment:
@poojahpatil90 `this.clickAwayHandler(…` doesn't seem to return anything, would we be able to replace:
```
this.connectionDomHandle.addEventListener('click', this.clickAwayHandler(this.connectionDomHandle), false)
```

with:

```
this.clickAwayHandler(this.connectionDomHandle)
```

? 😸

Reply:
Done!